pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -351,11 +353,23 @@ func (ssr *Snapshotter) CollectEventsSincePrevSnapshot(stopCh <-chan struct{}) (
 	}
 	lastEtcdRevision := resp.Header.Revision
 
+	metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindFull}).Set(0)
+	metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindDelta}).Set(0)
+
+	// if etcd revision newer than latest full snapshot revision,
+	// set `required` metric for full snapshot to 1
+	if ssr.PrevFullSnapshot == nil || ssr.PrevFullSnapshot.LastRevision != lastEtcdRevision {
+		metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindFull}).Set(1)
+	}
 	if ssr.prevSnapshot.LastRevision == lastEtcdRevision {
 		ssr.logger.Infof("No new events since last snapshot. Skipping initial delta snapshot.")
 		return false, nil
 	}
 
+	// need to take a delta snapshot here, because etcd revision is
+	// newer than latest snapshot revision
+	metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindDelta}).Set(1)
+
```

Comment:
Also, mark full snapshot required to true right?

Reply:
Yes, thanks